api/server/server.go

```diff
@@ -37,33 +39,72 @@ func Server(portNum int, taskPool *WorkerCont.TaskHandler, contextStruct *vms.In
 		encoder.Encode(result)
 	})
 
-	http.HandleFunc("POST /CreateVM", func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != http.MethodGet {
+	http.HandleFunc("/CreateVM", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost { // POST로 요청 제한
 			http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 			return
 		}
+
+		// TaskControlCreateVM 생성 및 요청 파싱
 		workerControl := &WorkerCont.TaskControlCreateVM{
-			ResultChann: make(chan WorkerCont.TaskExecutionResult),
+			ResultChan: make(chan string),
+			Vms:        contextStruct,
 		}
-		resultChannel := workerControl.ResultChann
+		resultChannel := workerControl.ResultChan
 		defer close(resultChannel)
-		workerControl.TaskUnparsor(r)
 
+		// JSON 파싱 및 에러 처리
+		if err := workerControl.TaskUnparsor(r); err != nil {
+			http.Error(w, "Failed to parse JSON", http.StatusBadRequest)
+			fmt.Printf("Error in TaskUnparsor: %v\n", err) // 필요 시 유지
+			return
+		}
+
+		// Task 생성 및 작업 할당
 		newTask := &WorkerCont.Task{
 			FunctionName: WorkerCont.CreateV,
 			TaskSpecific: workerControl,
 		}
-
 		taskPool.WorkerAllocate(newTask)
+
+		// 결과 처리 및 응답
 		result := <-resultChannel
 		encoder := json.NewEncoder(w)
-		encoder.Encode(result)
+		if err := encoder.Encode(result); err != nil {
+			http.Error(w, "Failed to encode result", http.StatusInternalServerError)
+			return
+		}
 	})
 
-	http.HandleFunc("GET /DeleteVM", func(w http.ResponseWriter, b *http.Request) {
-		taskPool.WorkerAllocate(&WorkerCont.Task{
+	http.HandleFunc("/DeleteVM", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
`DeleteVM` 같이 side effect가 있는 엔드포인트는 GET으로 접근할 수 있어서는 안 돼요. 사용자가 의도하지 않더라도 단순히 URL을 클릭하는 행위나 심지어 메신저 앱 같은 곳에서 미리보기를 보는 행위만으로도 DeleteVM을 실행하도록 할 수 있거든요. 단순히 정보를 얻는 엔드포인트가 아닌 경우 반드시 POST 같은 메서드를 붙여주세요!

이건 제가 미리 확인했어야 했는데 놓쳤었네요,,,!

```suggestion
	http.HandleFunc("POST /DeleteVM", func(w http.ResponseWriter, r *http.Request) {
```

Reply:
넵넵!! 고쳐 놓겠습니당!!